packages/create-react-app-demo/src/docs-chat/ai.tsx

```diff
@@ -1,32 +1,50 @@
 /** @jsx AI.createElement */
 /** @jsxFrag AI.Fragment */
 import * as AI from 'ai-jsx';
-import { DocsQA, ScoredChunk, LocalCorpus, defaultChunker, staticLoader } from 'ai-jsx/batteries/docs';
-// @ts-expect-error
-import markdownPath from './brand-new.md';
+import { DocsQA, DefaultFormatter, LocalCorpus, defaultChunker, staticLoader } from 'ai-jsx/batteries/docs';
 
-// For now, we load the ai.jsx docs from a local markdown file. Once we have a HTML->Markdown converter,
-// we can load the docs from the site directly.
-const docResponse = await fetch(markdownPath);
-const docText = await docResponse.text();
-const docs = [
-  {
-    pageContent: [docText],
-    name: 'Guide for AI Newcomers',
-  },
-];
-const corpus = new LocalCorpus(staticLoader(docs), defaultChunker);
+let globalCorpus: LocalCorpus | undefined;
 
-const corpusLoadedPromise = corpus.startLoading();
-
-const ShowDoc = ({ doc }: { doc: ScoredChunk }) => (
-  <>
-    Title: {doc.chunk.documentName ?? 'Untitled'}
-    Content: {doc.chunk.content}
-  </>
-);
+/**
+ * This is a very simple example of how to build a document corpus.
+ * We pull markdown content from a set of URLs and then index it.
+ */
+export async function indexCorpus() {
+  const files = {
+    'getting-started.md': 'Getting Started',
+    'guides/ai-ui.md': 'AI + UI',
+    'guides/brand-new.md': 'Guide for AI Newcomers',
+    'guides/docsqa.md': 'Docs QA: Grounding Answers',
+    'guides/esm.md': 'ESM',
+    'guides/jsx.md': 'JSX: Build System Considerations',
+    'guides/observability.md': 'Observability',
+    'guides/prompting.md': 'Getting the AI to say things',
+    'guides/rules-of-jsx.md': 'Rules of AI.JSX',
+  };
+  const docs = await Promise.all(
+    Object.entries(files).map(async ([path, title]) => {
+      const url = `https://raw.githubusercontent.com/fixie-ai/ai-jsx/main/packages/docs/docs/${path}`;
+      const response = await fetch(url);
+      const markdown = await response.text();
+      console.log(`Retrieved document from ${url}`);
+      return {
+        pageContent: [markdown],
+        name: title,
+      };
+    })
+  );
+  const corpus = new LocalCorpus(staticLoader(docs), defaultChunker);
+  const stats = await corpus.startLoading();
+  console.log(`Finished indexing documents, chunk count=${stats.numChunks}`);
+  return corpus;
+}
 
+/**
+ * Build the corpus on the first query, then use it for all subsequent queries.
+ */
 export async function DocsAgent({ question }: { question: string }) {
-  await corpusLoadedPromise;
-  return <DocsQA question={question} corpus={corpus} limit={5} docComponent={ShowDoc} />;
+  if (!globalCorpus) {
+    globalCorpus = await indexCorpus();
+  }
```

Comment:
Rather than this approach, I'd do:

```jsx
const indexCorpus = _.once(async function indexCorpus() {
  /* ... */
});

<DocsQA question={question} corpus={indexCorpus()} limit={5} docComponent={DefaultFormatter} />;
```

`_.once` creates a function where the wrapper is only invoked once; subsequent calls return the value returned by the first invocation.